internal/internal_workflow_client.go

```diff
@@ -496,3 +585,126 @@ func getRunID(runID string) *string {
+				return err1
+			}, serviceOperationRetryPolicy, isServiceTransientError)
+		if err != nil {
+			return nil, err
+		}
+		history.Events = append(history.Events, response.History.Events...)
+		if response.NextPageToken == nil {
+			break GetHistoryLoop
+		}
+		nexttoken = response.NextPageToken
+	}
+	return history, nil
+}
+
+func (iter *historyEventIteratorImpl) HasNext() bool {
+	return iter.nextEventIndex < len(iter.events) || len(iter.nexttoken) != 0 || iter.err != nil || !iter.initialized
+}
+
+func (iter *historyEventIteratorImpl) Next() (*s.HistoryEvent, error) {
+	// if caller call the Next() when iteration is over, just return nil, nil
+	if !iter.HasNext() {
+		return nil, nil
+	}
+
+	// we have cached events
+	if iter.nextEventIndex < len(iter.events) {
+		index := iter.nextEventIndex
+		iter.nextEventIndex++
+		return iter.events[index], nil
+	} else if iter.err != nil {
+		// we have err, clear that iter.err and return err
+		err := iter.err
+		iter.err = nil
+		return nil, err
+	}
+
+	iter.initialized = true
+	response, err := iter.paginate(iter.nexttoken)
+	iter.nextEventIndex = 0
+	if err == nil {
+		iter.events = response.History.Events
+		iter.nexttoken = response.NextPageToken
+		iter.err = nil
+	} else {
+		iter.events = nil
+		iter.nexttoken = nil
+		iter.err = err
+	}
+	return iter.Next()
+}
+
+func (workflowRun *workflowRunImpl) GetRunID() string {
+	return workflowRun.firstRunID
+}
+
+func (workflowRun *workflowRunImpl) Get(ctx context.Context, valuePtr interface{}) error {
+
+	iter := workflowRun.iterFn(ctx, workflowRun.currentRunID)
+	if !iter.HasNext() {
+		panic("could not get last history event for workflow")
+	}
+	closeEvent, err := iter.Next()
+	if err != nil {
+		return err
+	}
+
+	switch closeEvent.GetEventType() {
+	case s.EventTypeWorkflowExecutionCompleted:
+		attributes := closeEvent.WorkflowExecutionCompletedEventAttributes
+		if valuePtr == nil || attributes.Result == nil {
+			return nil
+		}
+		rf := reflect.ValueOf(valuePtr)
+		if rf.Type().Kind() != reflect.Ptr {
+			return errors.New("value parameter is not a pointer")
+		}
+		err = deSerializeFunctionResult(workflowRun.workflowFn, attributes.Result, valuePtr)
+	case s.EventTypeWorkflowExecutionCanceled:
+		attributes := closeEvent.WorkflowExecutionCanceledEventAttributes
+		err = NewCanceledError(attributes.Details)
+	case s.EventTypeWorkflowExecutionTerminated:
+		err = newTerminatedError()
+	case s.EventTypeWorkflowExecutionTimedOut:
+		attributes := closeEvent.WorkflowExecutionTimedOutEventAttributes
+		err = NewTimeoutError(attributes.GetTimeoutType())
+	case s.EventTypeWorkflowExecutionContinuedAsNew:
+		attributes := closeEvent.WorkflowExecutionContinuedAsNewEventAttributes
+		workflowRun.currentRunID = attributes.GetNewExecutionRunId()
+		return workflowRun.Get(ctx, valuePtr)
+	default:
+		err = errors.New("Unknown error occur when fetching workflow execution result")
```

Comment:
Could you provide more descriptive message (like the type of event returned instead of expected)

Reply:
changed to:
err = fmt.Errorf("Unexpected event type %s when handling workflow execution result", closeEvent.GetEventType())